controller/source/rocksdb_server/server.cpp

```diff
@@ -20,44 +27,42 @@ class session : public std::enable_shared_from_this<session> {
   session(tcp::socket socket, std::shared_ptr<rocksdb_proxy> rocksdb_proxy)
       : m_socket(std::move(socket))
       , m_rocksdb_proxy(std::move(rocksdb_proxy))
+      , m_strand(io_service)
   {
   }
 
   void start() {
-    handle_read();
+    auto self = shared_from_this();
+    m_strand.post([self]() {self->handle_read();});
   }
 
 private:
-  // handle_read() makes use of boost asio and it needs to call itself again to process the next chunk received.
-  // NOLINTBEGIN(misc-no-recursion)
   void handle_read() {
```

Comment:
Is there any way that the `handle_read()` function never terminates? e.g. if no EOF is sent and the connection remains open. I think this might be tricky.